### ISSUE TITLE
Support JS Design Documents

### ISSUE THREAD

**Issue by jedrichards** [glynnbird/couchmigrate#1]
Do you think it makes sense to support importing of valid JavaScript design documents too? That way the design documents could take advantage of linting or other automated checks that happen to the rest of your source code. Additionally it would make the formatting of functions less tiresome.

``` js
module.exports = {
  _id: '_design/fetch',
  views: {
    byx: {
      map: (function (doc) {
        emit(doc.x, doc.y);
      }).toString(),
      reduce: '_stats'
    }
  },
  language: 'javascript'
}
```

``` sh
couchmigrate --dd dd.js --db mydatabase
```

Or do you handle this some other way? Would be happy to put together a PR ...

EDIT:

Or indeed could pave the way for patterns like the below if you had complex validation functions,

``` js
var validate = require('./validate.js');

module.exports = {
  validate_doc_update: validate.toString(),
  _id: '_design/fetch',
  views: {
    byx: {
      map: (function (doc) {
        emit(doc.x, doc.y);
      }).toString(),
      reduce: '_stats'
    }
  },
  language: 'javascript'
}
```

(Oops clicked close by mistake).

**Comment by glynnbird** [comment]
Fixed in 0.1.0. Thanks to PR from @asmod3us :)

**Comment by jedrichards** [comment]
😁
